### ISSUE TITLE
Unable to start heron-tracker

### ISSUE THREAD

**Issue by kartik894** [twitter/heron#887]
Hi,

I have installed Apache Aurora in my cluster. I have tested Aurora by submitting jobs, and seems to work fine. However, I get the following error when starting the heron tracker. There are no issues with submission of topologies.

```
$ heron-tracker
08 Jun 2016 15:13:47-INFO:main.py:183: Running on port: 8888
08 Jun 2016 15:13:47-INFO:main.py:184: Using config file: /root/.herontools/conf/heron_tracker.yaml
08 Jun 2016 15:13:47-INFO:statemanagerfactory.py:61: Connecting to zk hostport: localhost:2181 rootpath: /heron
08 Jun 2016 15:13:47-INFO:connection.py:566: Connecting to localhost:2181
08 Jun 2016 15:13:47-INFO:connection.py:276: Sending request(xid=None): Connect(protocol_version=0, last_zxid_seen=0, time_out=10000, session_id=0, passwd='\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00', read_only=None)
08 Jun 2016 15:13:47-INFO:client.py:378: Zookeeper connection established, state: CONNECTED
08 Jun 2016 15:13:47-INFO:statemanagerfactory.py:81: Connecting to file state with rootpath: /root/.herondata/repository/state/local
08 Jun 2016 15:13:47-INFO:zkstatemanager.py:93: Adding children watch for path: /heron/topologies
08 Jun 2016 15:13:47-INFO:connection.py:276: Sending request(xid=1): GetChildren(path='/heron/topologies', watcher=<bound method ChildrenWatch._watcher of <kazoo.recipe.watchers.ChildrenWatch object at 0x7f04062f1350>>)
08 Jun 2016 15:13:47-INFO:connection.py:344: Received error(xid=1) NoNodeError((), {})
Traceback (most recent call last):
  File "/root/bin/heron-tracker/.bootstrap/_pex/pex.py", line 319, in execute
  File "/root/bin/heron-tracker/.bootstrap/_pex/pex.py", line 254, in _wrap_coverage
  File "/root/bin/heron-tracker/.bootstrap/_pex/pex.py", line 286, in _wrap_profiling
  File "/root/bin/heron-tracker/.bootstrap/_pex/pex.py", line 362, in _execute
  File "/root/bin/heron-tracker/.bootstrap/_pex/pex.py", line 420, in execute_entry
  File "/root/bin/heron-tracker/.bootstrap/_pex/pex.py", line 425, in execute_module
  File "/usr/lib/python2.7/runpy.py", line 180, in run_module
    fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/root/bin/heron-tracker/heron/tracker/src/python/main.py", line 194, in <module>
  File "/root/bin/heron-tracker/heron/tracker/src/python/main.py", line 189, in main
  File "/root/bin/heron-tracker/heron/tracker/src/python/main.py", line 42, in __init__
  File "/root/bin/heron-tracker/heron/tracker/src/python/tracker.py", line 79, in synch_topologies
  File "/root/bin/heron-tracker/heron/statemgrs/src/python/zkstatemanager.py", line 80, in get_topologies
  File "/root/bin/heron-tracker/heron/statemgrs/src/python/zkstatemanager.py", line 95, in _get_topologies_with_watch
  File "/root/.pex/install/kazoo-1.3.1-py2.7.egg.9e0018919913c23864670dfb4abcbd86b8867ea6/kazoo-1.3.1-py2.7.egg/kazoo/recipe/watchers.py", line 273, in __call__
    self._get_children()
  File "/root/.pex/install/kazoo-1.3.1-py2.7.egg.9e0018919913c23864670dfb4abcbd86b8867ea6/kazoo-1.3.1-py2.7.egg/kazoo/recipe/watchers.py", line 22, in wrapper
    return func(*args, **kwargs)
  File "/root/.pex/install/kazoo-1.3.1-py2.7.egg.9e0018919913c23864670dfb4abcbd86b8867ea6/kazoo-1.3.1-py2.7.egg/kazoo/recipe/watchers.py", line 283, in _get_children
    self._path, self._watcher)
  File "/root/.pex/install/kazoo-1.3.1-py2.7.egg.9e0018919913c23864670dfb4abcbd86b8867ea6/kazoo-1.3.1-py2.7.egg/kazoo/retry.py", line 122, in __call__
    return func(*args, **kwargs)
  File "/root/.pex/install/kazoo-1.3.1-py2.7.egg.9e0018919913c23864670dfb4abcbd86b8867ea6/kazoo-1.3.1-py2.7.egg/kazoo/client.py", line 945, in get_children
    return self.get_children_async(path, watch, include_data).get()
  File "/root/.pex/install/kazoo-1.3.1-py2.7.egg.9e0018919913c23864670dfb4abcbd86b8867ea6/kazoo-1.3.1-py2.7.egg/kazoo/handlers/threading.py", line 107, in get
    raise self._exception
kazoo.exceptions.NoNodeError: ((), {})
08 Jun 2016 15:13:50-ERROR:connection.py:558: Unhandled exception in connection loop
Traceback (most recent call last):
  File "/root/.pex/install/kazoo-1.3.1-py2.7.egg.9e0018919913c23864670dfb4abcbd86b8867ea6/kazoo-1.3.1-py2.7.egg/kazoo/protocol/connection.py", line 526, in _connect_attempt
    self._send_ping(connect_timeout)
  File "/root/.pex/install/kazoo-1.3.1-py2.7.egg.9e0018919913c23864670dfb4abcbd86b8867ea6/kazoo-1.3.1-py2.7.egg/kazoo/protocol/connection.py", line 445, in _send_ping
    self._submit(PingInstance, connect_timeout, PING_XID)
  File "/root/.pex/install/kazoo-1.3.1-py2.7.egg.9e0018919913c23864670dfb4abcbd86b8867ea6/kazoo-1.3.1-py2.7.egg/kazoo/protocol/connection.py", line 272, in _submit
    if request.type:
AttributeError: 'NoneType' object has no attribute 'type'

```

**Comment by javi1104** [comment]
Note that Tracker does not create any zookeeper nodes itself. It is a readonly service. If you launch the Tracker without creating /topologies node under the rootpath, Tracker will fail with a NoNodeError.

http://twitter.github.io/heron/docs/operators/heron-tracker/

**Comment by kramasamy** [comment]
@javi1104 - this has been fixed in 0.14.3 pre-release. Heron tracker polls the zk until some topologies come up. You are right in that it is a ready only service!
